### ISSUE TITLE
electron-updater fails to install update on MacOS

### ISSUE THREAD

**Issue by robertpatrick** [electron-userland/electron-builder#6396]
* **Electron-Builder Version**:  22.13.1

* **Node Version**: 16.13.0

* **Electron Version**: 13.5.2
* **Electron Type (current, beta, nightly)**: current

* **Electron-Updater Version**: 4.3.9

* **Target**: MacOS (default)

We are trying to release our application using electron-updater.  Everything works on Windows (other than we are also seeing the behavior described by #6329) and Linux but our MacOS app is failing to install the update.  The relevant section of the log looks like this:

```
info: Found version 0.8.13 (url: WebLogic-Kubernetes-Toolkit-UI-0.8.13-mac.zip, WebLogic-Kubernetes-Toolkit-UI-0.8.13.dmg)
info: Downloading update from WebLogic-Kubernetes-Toolkit-UI-0.8.13-mac.zip, WebLogic-Kubernetes-Toolkit-UI-0.8.13.dmg
debug: updater cache dir: /Users/rpatrick/Library/Application Support/Caches/wktui-updater
info: New version 0.8.13 has been downloaded to /Users/rpatrick/Library/Application Support/Caches/wktui-updater/pending/WebLogic-Kubernetes-Toolkit-UI-0.8.13-mac.zip
info: Download complete, install type: now
info: / requested
info: /1636044311994-1201.zip requested
info: /1636044311994-1201.zip requested by Squirrel.Mac, pipe /Users/rpatrick/Library/Application Support/Caches/wktui-updater/pending/WebLogic-Kubernetes-Toolkit-UI-0.8.13-mac.zip
warn: Code signature at URL file:///Users/rpatrick/Library/Caches/com.oracle.weblogic.wktui.ShipIt/update.UJFc5o2/WebLogic%20Kubernetes%20Toolkit%20UI.app/ did not pass validation: code object is not signed at all {"code":-1,"domain":"SQRLCodeSignatureErrorDomain"}
error: Error: Error: Code signature at URL file:///Users/rpatrick/Library/Caches/com.oracle.weblogic.wktui.ShipIt/update.UJFc5o2/WebLogic%20Kubernetes%20Toolkit%20UI.app/ did not pass validation: code object is not signed at all
```

The directory `/Users/rpatrick/Library/Caches/com.oracle.weblogic.wktui.ShipIt` is empty so the error message seems to be misleading.  I have no idea why this is failing but I am down to the last week before our release so any clues would be very welcome!

You can find our test releases at [https://github.com/oracle/weblogic-toolkit-ui/releases](https://github.com/oracle/weblogic-toolkit-ui/releases).

**Comment by mmaietta** [comment]
Hi there!
Would you mind trying with electron-updater@4.6.1? There's been a lot of changes since 4.3.9, and having 4.6.1 as a base version to work from will help.
Regarding the empty directory. Did this file at least exist or did the download/pipe fail entirely? `/Users/rpatrick/Library/Application Support/Caches/wktui-updater/pending/WebLogic-Kubernetes-Toolkit-UI-0.8.13-mac.zip`

**Comment by robertpatrick** [comment]
@mmaietta i will try with the newer version.  The file is downloaded to ~/Library/Application Support/Caches/wktui-updater/pending/.  I see three files created in the ~/Library/Caches/<bundle-id> directory but nothing in ~/Library/Caches/<bundle-id>.ShipIt directory.

**Comment by robertpatrick** [comment]
I was able to figure out how to fix the problem.  Due to the process requirements of my employer, we were unable to use electron-builder to sign the app and we cannot use the electron-publisher functionality to publish the release due to lack of support for a proxy server.  I tracked down a couple of problems related to permissions on the files and a missing zip.blockmap file not being attached to the release.  Addressing these issues resolved the problem--even with the 4.3.9 version.
